src/BookNavigator/book-navigator.js

```diff
@@ -103,58 +126,75 @@ export class BookNavigator extends LitElement {
    * to keep it in sync.
    */
   initializeBookSubmenus() {
-    const isBookProtected = this.bookreader.options.protected;
-    this.menuProviders = {
-      search: new SearchProvider(
+    const providers = {
+      downloads: new DownloadProvider(this.baseProviderConfig),
+      share: new SharingProvider(this.baseProviderConfig),
+      visualAdjustments: new VisualAdjustmentProvider({
+        ...this.baseProviderConfig,
+        /** Update menu contents */
+        onProviderChange: () => {
+          this.updateMenuContents();
+        },
+      }),
+    };
+
+    if (this.bookreader.options.enableSearch) {
```

Comment:
Since you've now got this conditionals for each of these providers (and they're instantiation is somewhat long).. I might break this down a bit for readability.. ie:
```
const options = this.bookreader.options;
if (options.enableSearch) {
  providers.search = this.getSearchProvider();
}
if (options.enableBookmarks) {
  providers.bookmarks = this.getBookmarksProvider();
}
...
getSearchProvider() {
  return new SearchProvider(...);
}
```

Reply:
thanks for the suggestion. drini and i talked about this, and since we are moving to bridge the plugins to their book-nav providers, we will keep the messy if blocks in their current position. 👍